Exercise 5: Basic Observation Skills in Data Analysis

Task:

- Load the How Much Sleep Do Americans Really Get? dataset into Jupyter.

- Identify columns that could be interesting for a specific type of analysis (e.g., trend analysis, group comparison) and explain your choice.


In [6]:
import pandas as pd

In [2]:
df_american_sleep = pd.read_csv('american_sleep.csv')

print("Dataset loaded successfully!")

Dataset loaded successfully!


In [3]:
df_american_sleep.info()

<class 'pandas.DataFrame'>
RangeIndex: 945 entries, 0 to 944
Data columns (total 9 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   index                     945 non-null    int64  
 1   Year                      945 non-null    int64  
 2   Period                    945 non-null    str    
 3   Avg hrs per day sleeping  945 non-null    float64
 4   Standard Error            945 non-null    float64
 5   Type of Days              945 non-null    str    
 6   Age Group                 945 non-null    str    
 7   Activity                  945 non-null    str    
 8   Sex                       945 non-null    str    
dtypes: float64(2), int64(2), str(5)
memory usage: 66.6 KB


In [4]:
df_american_sleep.head()


,index,Year,Period,Avg hrs per day sleeping,Standard Error,Type of Days,Age Group,Activity,Sex
0,0,2003,Annual,8.57,0.018,All days,15 years and over,Sleeping,Both
1,1,2004,Annual,8.55,0.026,All days,15 years and over,Sleeping,Both
2,2,2005,Annual,8.62,0.023,All days,15 years and over,Sleeping,Both
3,3,2006,Annual,8.63,0.024,All days,15 years and over,Sleeping,Both
4,4,2007,Annual,8.57,0.024,All days,15 years and over,Sleeping,Both


Analysis:

In [10]:
df_american_sleep.describe()

,index,Year,Avg hrs per day sleeping,Standard Error
count,945.000000,945.000000,945.000000,945.000000
mean,472.000000,2010.000000,8.806868,0.086958
std,272.942302,4.322782,0.545001,0.034631
min,0.000000,2003.000000,7.720000,0.018000
25%,236.000000,2006.000000,8.360000,0.063000
50%,472.000000,2010.000000,8.810000,0.084000
75%,708.000000,2014.000000,9.190000,0.106000
max,944.000000,2017.000000,10.650000,0.228000


Mean: The average American in this dataset sleeps about 8.80 hours.

Min/Max: Some people sleep 7.72 hours, while others sleep up to 10.65 hours.

Year: The data covers a range from 2003 to 2017.



In [12]:
df_american_sleep['Sex'].unique()



<StringArray>
['Both', 'Men', 'Women']
Length: 3, dtype: str

In [13]:
df_american_sleep['Age Group'].unique()

<StringArray>
['15 years and over',    '15 to 24 years',    '25 to 34 years',
    '35 to 44 years',    '45 to 54 years',    '55 to 64 years',
 '65 years and over']
Length: 7, dtype: str

Group Comparison and Trend Analysis:

Columns: Sex, Age Group, and Avg hrs per day sleeping.

I will focus specifically on the 'Men' and 'Women' categories to observe the differences in their sleep patterns

Explanation: I chose Sex because ".unique()" showed that the data is divided into 'Men' and 'Women,' which is perfect for a Group Comparison to see who sleeps more.

I chose Age Group because it allows for a Trend Analysis to see if sleep changes as people move from '15-24 years' to '65 years and over.'

I chose Avg hrs per day sleeping because my .describe() results showed a wide range of sleep times (from 7.7 to 10.6 hours), and I want to see how that variety is distributed among the different groups.

In [14]:

women_sleep = df_american_sleep[df_american_sleep['Sex'] == 'Women']
print("Average sleep for Women:", women_sleep['Avg hrs per day sleeping'].mean())

men_sleep = df_american_sleep[df_american_sleep['Sex'] == 'Men']
print("\nAverage sleep for Men:", men_sleep['Avg hrs per day sleeping'].mean())

Average sleep for Women: 8.881174603174602

Average sleep for Men: 8.7315873015873
